service/matching/tasklist/matcher.go

```diff
@@ -429,7 +429,11 @@ func (tm *taskMatcherImpl) Poll(ctx context.Context, isolationGroup string) (*In
 		TaskListKind: tm.tasklistKind.Ptr(),
 		EventName:    "Matcher Falling Back to Non-Local Polling",
 	})
-	return tm.pollOrForward(ctxWithCancelPropagation, startT, isolationGroup, isolatedTaskC, tm.taskC, tm.queryTaskC)
+	task, err := tm.pollOrForward(ctxWithCancelPropagation, startT, isolationGroup, isolatedTaskC, tm.taskC, tm.queryTaskC)
+	if task != nil {
+		task.PollMatchLatency = time.Since(startT)
```

Comment:
This is poll latency which is different than task latency. Task might have been sitting in matching for longer. Poll latency could easily be calculated on the worker side as well.

Reply:
cc: @shijiesheng

Reply:
We want the poller wait time on the matching engine. This might be more accurate since it excludes all the other calls.

Reply:
Synced offline, more specifically we want to exclude the latency of updating history before returning task to poller. Therefore calculating poller wait time within the engine is needed to achieve that level of accuracy.